### PR TITLE
Add a select to choose the percentage of people respecting the auto-isolation rule

### LINKED ISSUE #7: Feature: people arbitrarily choose whether to isolate themselves or not
Now when infected ball A collides with not-infected ball B, then the app notifies ball B that stops moving.
In a real world, someone can choose whether to isolate themselves or not, so I would add an option to the sidepanel to choose the % of people that will follow the auto-isolation rule.

### PR DESCRIPTION
**Pull request by gioid**
Closed #7

### CODE REVIEW

File: src/Ball.js [hunk 1]
```diff
@@ -44,7 +47,7 @@ export class Ball {
           return
         }
       }
-      if (this.hasAppInstalled) {
+      if (this.hasAppInstalled && this.willRespectAutoIsolation) {
```
Comment (brushtyler) [inline src/Ball.js]: I think this if block must be removed:
it is quite similar to the one on line 113, with the difference that line 113 also checks for `otherBall.hasAppInstalled` (i.e. if both have the app installed).

File: src/Ball.js [hunk 2]
```diff
@@ -114,7 +119,7 @@ export class Ball {
             otherBall.state = STATES.incubating
             RUN.results[STATES.incubating]++
             RUN.results[STATES.well]--
-            if (this.isAwareToBeInfected(otherBall)) {
+            if (this.isAwareToBeInfected(otherBall) && this.willRespectAutoIsolation) {
```
Comment (brushtyler) [inline src/Ball.js]: I think this check needs to be changed as it stops movements for `otherBall` when `this` ball will respect autoisolation.
```suggestion
            if (otherBall.isAwareToBeInfected(this) && otherBall.willRespectAutoIsolation) {
```

File: src/Ball.js [hunk 3]
```diff
@@ -30,6 +31,8 @@ export class Ball {
     this.hasCollision = true
     this.survivor = false
     this.hasAppInstalled = hasAppInstalled
+    // a person can arbitrarily choose whether to respect auto-isolation or not
+    this.willRespectAutoIsolation = sketch.random(100) <= peopleRespectingAutoIsolationPercentage
```
Comment (marcopierobon) [inline src/Ball.js:35]: Are we planning to represent these?

**Review by brushtyler — CHANGES_REQUESTED** [review]

File: src/Ball.js (hunk 3 above)
Comment (gioid) [reply src/Ball.js:35]: I think that can be useful to understand the importance to respect auto-isolation. What do you think about it?

File: src/Ball.js (hunk 1 above)
Comment (gioid) [reply src/Ball.js]: I agree

File: src/Ball.js (hunk 2 above)
Comment (gioid) [reply src/Ball.js]: Absolutely yes!

File: src/Ball.js (hunk 1 above)
Comment (gioid) [reply src/Ball.js]: Fixed with the last commit

File: src/Ball.js (hunk 2 above)
Comment (gioid) [reply src/Ball.js]: Fixed with the last commit.

**Review by brushtyler — APPROVED** [review]

File: src/index.html [hunk 4]
```diff
@@ -388,6 +388,23 @@ <h1>COVID-19 Spread Simulator 🦠</h1>
               </select>
             </span>
           </div>
+          <div class="parameterField">
+            <div class="parameterTextLine tooltip ">
+              Conformists (%)
```
Comment (brushtyler) [inline src/index.html]: As suggested by @marcopierobon on slack, it would be better to replace "Conformists" with "Self-isolating"

Comment (gioid) [reply src/index.html]: Done

**Pull request merged**